src/include/function/aggregate/avg.h

```diff
@@ -80,7 +81,12 @@ struct AvgFunction {
     static void finalize(uint8_t* state_) {
         auto state = reinterpret_cast<AvgState*>(state_);
         if (!state->isNull) {
-            state->avg = state->sum / (double_t)state->count;
+            if constexpr (std::is_same_v<T, common::int128_t>) {
```

Comment:
Use template specialization instead of `if constexpr` to improve readability.
Also add a test case on avg function on huge int.